qiskit/chemistry/ground_state_calculation/mes_factories/vqe_uccsd_factory.py

```diff
@@ -0,0 +1,59 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The minimum eigensolver factory for ground state calculation algorithms."""
+
+from qiskit.aqua import QuantumInstance
+from qiskit.aqua.algorithms import MinimumEigensolver, VQE
+from qiskit.chemistry.components.variational_forms import UCCSD
+from qiskit.chemistry.qubit_transformations import QubitOperatorTransformation
+from qiskit.chemistry.components.initial_states import HartreeFock
+
+from .mes_factory import MESFactory
+
+
+class VQEUCCSDFactory(MESFactory):
```

Comment:
I still really question the utility of this over a callback. This class, without exposing any config of VQE etc is so 'static'. And for a single method its so clunky for a user to have to wrap their solver in a class instance for one method.

Reply:
Let's keep the discussion about that here: https://github.com/Qiskit/qiskit-aqua/pull/1288#discussion_r496876071